pinterest/teletraan
# Don't pause deployment on new instance from ASG scaling

Currently, when deployment is paused, Teletraan stops installing service on all the hosts part of the cluster. Deployment is also paused/disabled on any new hosts came up due to ASG scaling. This makes pause a very risky operation. For example, pausing sidecar deploy would cause clusters to not get that sidecar.


Fix: Pausing a deploy will only be effective on existing hosts. New hosts (from ASG scale up) will continue installing most recent version that's deployed or being deployed.

Tests:
1. Verify that service is not installed on existing hosts when deployment is paused.
2. Scale up ASG and verify that service is installed on new hosts when deployment is paused.
3. Verify service continue to install on existing hosts when deployment is resumed.
